applications/FluidDynamicsApplication/custom_utilities/estimate_dt_utilities.h

```diff
@@ -264,42 +156,48 @@ class EstimateDtUtility
     double    mDtMin;       // User-defined minimum time increment allowed
     ModelPart &mrModelPart; // The problem's model part
 
-    ///@} // Member variables
+	///@}
+	///@name Serialization
+	///@{
+
+	friend class Serializer;
+
+	void save(Serializer& rSerializer) const;
+
+	void load(Serializer& rSerializer);
+
+    ///@}
     ///@name Private Operations
     ///@{
 
-    static double CalculateElementCFL(Element &rElement, GeometryDataContainer& rGeometryInfo, double Dt)
-    {
-        double Proj = 0.0;
-
-        // Get the element's geometric parameters
-        const auto& r_geometry = rElement.GetGeometry();
-        GeometryUtils::CalculateGeometryData(r_geometry, rGeometryInfo.DN_DX, rGeometryInfo.N, rGeometryInfo.Area);
-
-        // Elemental Velocity
-        array_1d<double,3> ElementVel = rGeometryInfo.N[0]*r_geometry[0].FastGetSolutionStepValue(VELOCITY);
-        for (unsigned int i = 1; i < TDim+1; ++i)
-            ElementVel += rGeometryInfo.N[i]*r_geometry[i].FastGetSolutionStepValue(VELOCITY);
-
-        // Calculate u/h as the maximum projection of the velocity along element heights
-        for (unsigned int i = 0; i < TDim+1; ++i)
-        {
-            for (unsigned int d = 0; d < TDim; ++d)
-                Proj += ElementVel[d]*rGeometryInfo.DN_DX(i,d);
-            Proj = fabs(Proj);
-        }
-
-        return Proj*Dt;
-    }
-
+    /**
+     * @brief Calulate element CFL number
+     * For the given element, this method calculates the CFL number
+     * @param rElement Element to calculate the CFL number
+     * @param rGeometryInfo Auxiliary geometry data container
+     * @param Dt Current time increment
+     * @return double The element CFL number
+     */
+    static double CalculateElementCFL(
+        const Element &rElement,
+        const ElementSizeFunctionType& rElementSizeCalculator,
+        const double Dt);
 
+    /**
+     * @brief Get the minimum element size calculation function
+     * This method checks the geometry of the provided element and returns the corresponding
+     * minimum element size calculation fucntion.
+     * @param rGeometry Geoemtry in which the element size is to be computed
+     * @return ElementSizeFunctionType Function to calculate the minimum element size
+     */
+    static ElementSizeFunctionType GetMinimumElementSizeFunction(const Geometry<Node<3>>& rGeometry);
```

Comment:
since those are private you don't need them in the header I would say, you can completely declare this fct in the cpp

(Note that this is a matter of taste, I like to put as few things in headers as possible to minimize compile times)

Reply:
I prefer to keep all the functions definitions in the header. I think this simplifies when browsing the code.